### PR TITLE
feat(httpclient): A http client component for tinygo

### PR DESCRIPTION
**Pull request by ritesh089**
## Feature or Problem
Http client example for tinygo.

### CODE REVIEW

File: examples/golang/components/http-client-tinygo/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# HTTP Hello World
+
+This is a simple TinyGo Wasm example that makes accepts a http request , then makes a http call to an external endpoint and returns a response for the request.
```
Comment (brooksmtownsend) [inline examples/golang/components/http-client-tinygo/README.md:3]:
```suggestion
# HTTP Client Example

This is a simple TinyGo Wasm example that makes accepts a http request, then makes a http call to an external endpoint and returns a response for the request.
```

File: examples/golang/components/http-client-tinygo/hello.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	httpserver := HttpServer{}
+	// Set the incoming handler struct to HttpServer
+	http.SetExportsWasiHttp0_2_0_IncomingHandler(httpserver)
+}
+
+func (h HttpServer) Handle(request HttpRequest, responseWriter HttpResponseWriter) {
+	req := http.NewOutgoingRequest(http.NewFields())
+	req.SetScheme(http.Some[http.WasiHttp0_2_0_TypesScheme](http.WasiHttp0_2_0_TypesSchemeHttps()))
+	req.SetAuthority(http.Some("postman-echo.com"))
+	req.SetPathWithQuery(http.Some("/bytes/100/mb"))
```
Comment (brooksmtownsend) [inline examples/golang/components/http-client-tinygo/hello.go:28]: What do you think about cutting this down to a lower amount, like 10kb just so the terminal output is a little more manageable?

Our other example is the dog-fetcher, which could be a good way to give back an output to the user that they can see the result right away

File: examples/golang/components/http-client-tinygo/hello.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+			httpResponse.SetStatusCode(200)
+			body := httpResponse.Body().Unwrap()
+			bodyWrite := body.Write().Unwrap()
+			instream := responseBody.Stream().Unwrap()
+			okResponse := http.Ok[HttpOutgoingResponse, HttpError](httpResponse)
+			http.StaticResponseOutparamSet(responseWriter, okResponse)
+			splice(instream, bodyWrite)
+			bodyWrite.Drop()
+			http.StaticOutgoingBodyFinish(body, http.None[http.WasiHttp0_2_0_TypesTrailers]())
+		}
```
Comment (brooksmtownsend) [inline examples/golang/components/http-client-tinygo/hello.go:47]: I think here what might be better is set the outgoing response to the returned status code and reading the body even if the status code isn't 200, what do you think?

File: examples/golang/components/http-client-tinygo/wadm.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  components:
+    - name: http-component
+      type: component
+      properties:
+        image: file://./build/http_client_tinygo_s.wasm
+      traits:
+        # Govern the spread/scheduling of the component
+        - type: spreadscaler
+          properties:
+            replicas: 1
```
Comment (brooksmtownsend) [inline examples/golang/components/http-client-tinygo/wadm.yaml:18]: Here I think we need to add the http client capability provider and a link to it to make it work in wasmCloud

**Review by brooksmtownsend — CHANGES_REQUESTED** [review]
I think this example is really nice to show how to work with large HTTP payloads! I had a couple requests to improve the example/documentation

File: examples/golang/components/http-client-tinygo/hello.go (hunk 3 above)
Comment (ritesh089) [reply examples/golang/components/http-client-tinygo/hello.go:47]: so we want to return the body as is ? I can just remove the status code check in that case .

File: examples/golang/components/http-client-tinygo/hello.go (hunk 2 above)
Comment (ritesh089) [reply examples/golang/components/http-client-tinygo/hello.go:28]: sure . i ll change the endpoint.

File: examples/golang/components/http-client-tinygo/README.md (hunk 1 above)
Comment (ritesh089) [reply examples/golang/components/http-client-tinygo/README.md:3]: will update.

Comment (ritesh089) [reply examples/golang/components/http-client-tinygo/README.md]: Done

File: examples/golang/components/http-client-tinygo/hello.go (hunk 3 above)
Comment (ritesh089) [reply examples/golang/components/http-client-tinygo/hello.go:47]: I have made the changes as requested. Please review

File: examples/golang/components/http-client-tinygo/wadm.yaml (hunk 4 above)
Comment (ritesh089) [reply examples/golang/components/http-client-tinygo/wadm.yaml:18]: done

**Review by brooksmtownsend — APPROVED** [review]
```
➜ curl localhost:8080 | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    99    0    99    0     0    809      0 --:--:-- --:--:-- --:--:--   811
{
  "message": "https://images.dog.ceo/breeds/hound-basset/n02088238_9635.jpg",
  "status": "success"
}
```

Looks great! Look at that cute pup!

**Pull request merged**